src/gui/folderman.cpp

```diff
@@ -1394,90 +1397,62 @@ static QString canonicalPath(const QString &path)
     return selFile.canonicalFilePath();
 }
 
-QString FolderMan::checkPathValidityForNewFolder(const QString &path, const QUrl &serverUrl) const
+QString FolderMan::checkPathValidityForNewFolder(const QString &path) const
 {
-    QString recursiveValidity = checkPathValidityRecursive(path);
-    if (!recursiveValidity.isEmpty()) {
-        qCDebug(lcFolderMan) << path << recursiveValidity;
-        return recursiveValidity;
-    }
-
     // check if the local directory isn't used yet in another ownCloud sync
-    Qt::CaseSensitivity cs = Qt::CaseSensitive;
-    if (Utility::fsCasePreserving()) {
-        cs = Qt::CaseInsensitive;
-    }
+    const auto cs = Utility::fsCaseSensitivity();
 
     const QString userDir = QDir::cleanPath(canonicalPath(path)) + '/';
     for (auto i = _folderMap.constBegin(); i != _folderMap.constEnd(); ++i) {
         Folder *f = static_cast<Folder *>(i.value());
-        QString folderDir = QDir::cleanPath(canonicalPath(f->path())) + '/';
+        const QString folderDir = QDir::cleanPath(canonicalPath(f->path())) + '/';
 
-        bool differentPaths = QString::compare(folderDir, userDir, cs) != 0;
-        if (differentPaths && folderDir.startsWith(userDir, cs)) {
+        if (QString::compare(folderDir, userDir, cs) == 0) {
+            return tr("There is already a sync from the server to this local folder. "
+                      "Please pick another local folder!");
```

Comment:
The check checks for sync connection from any server, right? So maybe a better string would be something like `This local folder is already used for another sync. Please pick...`